agentendpoint/inventory.go

```diff
@@ -0,0 +1,272 @@
+package agentendpoint
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	"github.com/GoogleCloudPlatform/osconfig/attributes"
+	"github.com/GoogleCloudPlatform/osconfig/config"
+	agentendpointpb "github.com/GoogleCloudPlatform/osconfig/internal/google.golang.org/genproto/googleapis/cloud/osconfig/agentendpoint/v1alpha1"
+	"github.com/GoogleCloudPlatform/osconfig/inventory"
+	"github.com/GoogleCloudPlatform/osconfig/packages"
+	"github.com/golang/protobuf/ptypes"
+	timestamp "github.com/golang/protobuf/ptypes/timestamp"
+)
+
+const (
+	inventoryURL = config.ReportURL + "/guestInventory"
+	maxRetries   = 5
+)
+
+// ReportInventory reports inventory to agent endpoint and writes it to guest attributes.
+func (c *Client) ReportInventory(ctx context.Context) {
+	state := inventory.Get()
+	write(state, inventoryURL)
+	c.report(ctx, state)
+}
+
+func write(state *inventory.InstanceInventory, url string) {
+	logger.Debugf("Writing instance inventory to guest attributes.")
```

Comment:
Sorry to do this, but I just pushed new logging system, use clog now for all logging calls (will need to bring context into all your functions).

Reply:
If you merge master into development I'll pull and update the logging system